package/MDAnalysis/core/universe.py

```diff
@@ -369,9 +369,20 @@ def __init__(self, topology=None, *coordinates, all_coordinates=False,
             if callable(transformations):
                 transformations = [transformations]
             self._trajectory.add_transformations(*transformations)
-
+        to_guess = list(to_guess)
+        # add mass and type to the to_guess list
+        toplist = list(self._topology.read_attributes)
+        if not any(att.singular == 'type' for att in toplist) and 'types' not in to_guess:
+            to_guess.append('types')
+        if not any(att.singular == 'mass' for att in toplist) and 'masses' not in to_guess:
+            to_guess.append('masses')
+        if 'bonds' in to_guess:
```

Comment:
I  handled this by checking if the '`bonds`' are supplied in the 'to_guess' list then we will make '`guess_bonds`' value to be `True` and remove the bonds from the '`to_guess`' list to not cause a confusion

Reply:
I still think that `guess_bonds` should be deprecated. It is not good to have two ways, at the same place, to ask for the same thing. It can easily confuse the users. You can take care of the deprecation later, though.

Reply:
I agree with you indeed. Maybe I didn't express it clearly.